### DISCUSSION TITLE
No detections for today?

### DISCUSSION THREAD

**Discussion by dbrianhendrix** [mcguirepr89/BirdNET-Pi#427] [General]
It appears this is becoming a regular occurrence -- the system is staying very behind. I'm showing NO detections for today and the log is showing it working on recordings from yesterday evening. 

<img width="1515" alt="Screen Shot 2022-07-07 at 9 36 21 AM" src="https://user-images.githubusercontent.com/52499277/177801017-aa5dd2cf-6050-4d38-9131-22d000281535.png">

**Comment by dbrianhendrix** [comment]
<img width="1817" alt="Screen Shot 2022-07-07 at 9 51 59 AM" src="https://user-images.githubusercontent.com/52499277/177804345-f346e552-1b80-4573-ac69-1520707ee34e.png">

**Comment by Svardsten53** [comment]
At least two of us here in Sweden also had this problem earlier today. Restarting services helped and issue seems to be solved, This issue has been discussed in other threads as well.

**Comment by dbrianhendrix** [comment]
I rebooted the Pi but I'm still not caught up. Currently, I'm still 4 hours behind but keeping an eye on it.

**Reply by nilspupils** [reply]
Had the same problem yesterday, resolved with rebooting.

**Reply by grajohnt** [reply]
Came here to post a similar problem - neither rebooting nor restarting services helped. It's now 6 hours behind, but verrry slowly catching up. (Note, running a 3B+ here, and just enabled the zram swap).

I'm still very much in 'setup' mode with mine, so I could either manually delete the files to accelerate the processing, or alternatively blitz all of the data and start from scratch. Any less-radical suggestions?

**Comment by grajohnt** [comment]
Watching the log, I think this is going to be a recurring problem.

After some hours of 'catching up', the timing of processing past audio chunks suddenly leaps from 3-4s for each to 30s each. Once this happens, the queue will fall further and further behind, given that they're 15s clips. Restarting services (or rebooting) fixes the problem, but it seems to happen again regardless. (I'll screenshot my log if I catch it again).

Any idea what might be causing this?

**Reply by grajohnt** [reply]
Here we go - after four hours, it went from 3-4s up to much higher, and then hasn't recovered since - typicallying now at 5-8s per chunk, but spiking up higher from time to time. It _did_ catch up, and was live at this point.

19:01:04---ANALYZING AUDIO... DONE! Time 3.8 SECONDS
19:01:04---WRITING RESULTS TO BirdSongs/July-2022/09-Saturday/2022-07-09-birdnet-19:00:27.wav.csv ... DONE! WROTE 0 RESULTS.
19:01:10---READING AUDIO DATA... DONE! READ 5 CHUNKS.
19:01:10---Date-time: 2022-07-09 19:00:43
19:01:29---0.0;3.0-('Setophaga chrysoparia_Golden-cheeked Warbler', 0.025666306)
19:01:29---3.0;6.0-('Rallus elegans_King Rail', 0.03884067)
19:01:29---6.0;9.0-('Setophaga chrysoparia_Golden-cheeked Warbler', 0.024976796)
19:01:29---9.0;12.0-('Setophaga chrysoparia_Golden-cheeked Warbler', 0.028984409)
19:01:29---12.0;15.0-("Limnothlypis swainsonii_Swainson's Warbler", 0.02779412)
19:01:31---RECORDING_LENGTH set to 15
19:01:31---BirdNET-Pi/birdnet/bin/python3 BirdNET-Pi/scripts\analyze.py --i BirdSongs/July-2022/09-Saturday/2022-07-09-birdnet-19:00:59.wav --o BirdSongs/July-2022/09-Saturday/2022-07-09-birdnet-19:00:59.wav.csv --lat 30.XX --lon -95.XX --week 26 --overlap 0.0 --sensitivity 1.25 --min_conf 0.7 --exclude_listBirdNET-Pi/exclude_species_list.txt
19:01:32---ANALYZING AUDIO... DONE! Time 18.5 SECONDS
19:01:32---WRITING RESULTS TO BirdSongs/July-2022/09-Saturday/2022-07-09-birdnet-19:00:43.wav.csv ... DONE! WROTE 0 RESULTS.
19:01:39---READING AUDIO DATA... DONE! READ 5 CHUNKS.
19:01:39---Date-time: 2022-07-09 19:00:59
19:02:06---0.0;3.0-('Vireo atricapilla_Black-capped Vireo', 0.030214146)
19:02:06---3.0;6.0-("Vireo bellii_Bell's Vireo", 0.033778332)
19:02:06---6.0;9.0-('Setophaga chrysoparia_Golden-cheeked Warbler', 0.029996246)
19:02:06---9.0;12.0-('Vireo atricapilla_Black-capped Vireo', 0.025476333)
19:02:06---12.0;15.0-('Vireo atricapilla_Black-capped Vireo', 0.027640227)
19:02:11---Files loaded
19:02:11---RECORDING_LENGTH set to 15
19:02:12---BirdNET-Pi/birdnet/bin/python3 BirdNET-Pi/scripts\analyze.py --i BirdSongs/July-2022/09-Saturday/2022-07-09-birdnet-19:01:15.wav --o BirdSongs/July-2022/09-Saturday/2022-07-09-birdnet-19:01:15.wav.csv --lat 30.XX --lon -95.XX --week 26 --overlap 0.0 --sensitivity 1.25 --min_conf 0.7 --exclude_listBirdNET-Pi/exclude_species_list.txt
19:02:12---ANALYZING AUDIO... DONE! Time 27.2 SECONDS
19:02:12---WRITING RESULTS TO BirdSongs/July-2022/09-Saturday/2022-07-09-birdnet-19:00:59.wav.csv ... DONE! WROTE 0 RESULTS.
19:02:27---READING AUDIO DATA... DONE! READ 5 CHUNKS.
19:02:27---Date-time: 2022-07-09 19:01:15
19:03:05---0.0;3.0-('Setophaga chrysoparia_Golden-cheeked Warbler', 0.021699281)
19:03:05---3.0;6.0-("Vireo bellii_Bell's Vireo", 0.031925738)
19:03:05---6.0;9.0-('Setophaga chrysoparia_Golden-cheeked Warbler', 0.028132735)
19:03:05---9.0;12.0-('Myiarchus cinerascens_Ash-throated Flycatcher', 0.027751708)
19:03:05---12.0;15.0-('Dendragapus obscurus_Dusky Grouse', 0.02927581)
19:03:06---RECORDING_LENGTH set to 15
19:03:06---BirdNET-Pi/birdnet/bin/python3 BirdNET-Pi/scripts\analyze.py --i BirdSongs/July-2022/09-Saturday/2022-07-09-birdnet-19:01:30.wav --o BirdSongs/July-2022/09-Saturday/2022-07-09-birdnet-19:01:30.wav.csv --lat 30.XX --lon -95.XX --week 26 --overlap 0.0 --sensitivity 1.25 --min_conf 0.7 --exclude_listBirdNET-Pi/exclude_species_list.txt
19:03:07---ANALYZING AUDIO... DONE! Time 38.1 SECONDS
19:03:07---WRITING RESULTS TO BirdSongs/July-2022/09-Saturday/2022-07-09-birdnet-19:01:15.wav.csv ... DONE! WROTE 0 RESULTS.
19:03:12---READING AUDIO DATA... DONE! READ 5 CHUNKS.
19:03:12---Date-time: 2022-07-09 19:01:30
19:03:18---0.0;3.0-('Micrathene whitneyi_Elf Owl', 0.026411371)
19:03:18---3.0;6.0-('Setophaga chrysoparia_Golden-cheeked Warbler', 0.027976435)
19:03:18---6.0;9.0-('Catherpes mexicanus_Canyon Wren', 0.02252973)
19:03:18---9.0;12.0-('Setophaga chrysoparia_Golden-cheeked Warbler', 0.03330411)
19:03:18---12.0;15.0-('Dendragapus obscurus_Dusky Grouse', 0.027632724)
19:03:18---RECORDING_LENGTH set to 15
19:03:18---BirdNET-Pi/birdnet/bin/python3 BirdNET-Pi/scripts\analyze.py --i BirdSongs/July-2022/09-Saturday/2022-07-09-birdnet-19:01:46.wav --o BirdSongs/July-2022/09-Saturday/2022-07-09-birdnet-19:01:46.wav.csv --lat 30.XX --lon -95.XX --week 26 --overlap 0.0 --sensitivity 1.25 --min_conf 0.7 --exclude_listBirdNET-Pi/exclude_species_list.txt
19:03:19---ANALYZING AUDIO... DONE! Time 5.5 SECONDS
19:03:19---WRITING RESULTS TO BirdSongs/July-2022/09-Saturday/2022-07-09-birdnet-19:01:30.wav.csv ... DONE! WROTE 0 RESULTS.
19:03:23---READING AUDIO DATA... DONE! READ 5 CHUNKS.
19:03:23---Date-time: 2022-07-09 19:01:46
19:03:29---0.0;3.0-('Lanius ludovicianus_Loggerhead Shrike', 0.033737153)
19:03:29---3.0;6.0-('Setophaga chrysoparia_Golden-cheeked Warbler', 0.0404137)
19:03:29---6.0;9.0-('Lanius ludovicianus_Loggerhead Shrike', 0.06741395)
19:03:29---9.0;12.0-('Lanius ludovicianus_Loggerhead Shrike', 0.040651016)
19:03:29---12.0;15.0-('Lanius ludovicianus_Loggerhead Shrike', 0.036910355)
19:03:29---RECORDING_LENGTH set to 15
19:03:29---BirdNET-Pi/birdnet/bin/python3 BirdNET-Pi/scripts\analyze.py --i BirdSongs/July-2022/09-Saturday/2022-07-09-birdnet-19:02:02.wav --o BirdSongs/July-2022/09-Saturday/2022-07-09-birdnet-19:02:02.wav.csv --lat 30.XX --lon -95.XX --week 26 --overlap 0.0 --sensitivity 1.25 --min_conf 0.7 --exclude_listBirdNET-Pi/exclude_species_list.txt
19:03:30---ANALYZING AUDIO... DONE! Time 5.7 SECONDS
19:03:30---WRITING RESULTS TO BirdSongs/July-2022/09-Saturday/2022-07-09-birdnet-19:01:46.wav.csv ... DONE! WROTE 0 RESULTS.
19:03:34---READING AUDIO DATA... DONE! READ 5 CHUNKS.
19:03:34---Date-time: 2022-07-09 19:02:02
19:03:41---0.0;3.0-('Tyto alba_Barn Owl', 0.2043505)
19:03:41---3.0;6.0-('Tyto alba_Barn Owl', 0.09617661)
19:03:41---6.0;9.0-('Lanius ludovicianus_Loggerhead Shrike', 0.02986619)
19:03:41---9.0;12.0-('Lanius ludovicianus_Loggerhead Shrike', 0.029859787)
19:03:41---12.0;15.0-('Catherpes mexicanus_Canyon Wren', 0.03412814)
19:03:45---Files loaded
19:03:45---RECORDING_LENGTH set to 15
19:03:46---BirdNET-Pi/birdnet/bin/python3 BirdNET-Pi/scripts\analyze.py --i BirdSongs/July-2022/09-Saturday/2022-07-09-birdnet-19:02:17.wav --o BirdSongs/July-2022/09-Saturday/2022-07-09-birdnet-19:02:17.wav.csv --lat 30.XX --lon -95.XX --week 26 --overlap 0.0 --sensitivity 1.25 --min_conf 0.7 --exclude_listBirdNET-Pi/exclude_species_list.txt
19:03:46---ANALYZING AUDIO... DONE! Time 6.9 SECONDS
19:03:46---WRITING RESULTS TO BirdSongs/July-2022/09-Saturday/2022-07-09-birdnet-19:02:02.wav.csv ... DONE! WROTE 0 RESULTS.
19:03:51---READING AUDIO DATA... DONE! READ 5 CHUNKS.

**Reply by grajohnt** [reply]
I'm reasonably sure it would start falling further and further behind at this point without a restart of services.

**Reply by grajohnt** [reply]
FWIW, my 'fix' for the moment is adding reboots every four hours in crontab.

**Reply by snaterendesnader** [reply]
I have the same problem, it started a few days ago. Restarting services helps, but for some weird reason rebooting the Pi every 1.5hs doesn't solve the issue.

Since rebooting doesn't help, but restarting services does I think the problem lies with the processing of the backlog of recordings, as restarting the services cleans up the recordings.

**Reply by mcguirepr89** [reply]
Hello, all -- I am sorry for the relative silence while I've been out of town. I removed the section in the `cleanup.sh` script that checks for backlogged files to stop services in that event. You should be able to address this issue with an update:
"Tools" > "System Controls" > "Update"

**Reply by snaterendesnader** [reply]
Ah cool, I'm going to update and see how that goes, thanks!

**Reply by snaterendesnader** [reply]
@mcguirepr89 seems to be going well since this update, Birdnet has been running without problems during the night, no need to restart services.

**Reply by grajohnt** [reply]
Unfortunately not the same experience for me - I updated, and removed the reboots, and it fell behind again. (But admittedly, I'm running on a 3B+, which is pretty slow)

It was running fine with 4-hourly reboots, so I've returned to that!

**Reply by ehpersonal38** [reply]
Hmm, I'd be interested in seeing what `htop` shows about the CPU and memory when the spikes occur. Maybe some other program is using a lot of resources which causes the spike.

**Reply by grajohnt** [reply]
OK - I'll turn off some of the reboots this weekend and keep an eye on it and report back.

**Reply by grajohnt** [reply]
Caught it taking longer to process than it does to record - bounces around from about 7.5 to 60+ seconds to process each audio recording.

09:07:04---ANALYZING AUDIO... DONE! Time 26.9 SECONDS
09:07:04---WRITING RESULTS TO BirdSongs/July-2022/15-Friday/2022-07-15-birdnet-08:09:56.wav.csv ... DONE! WROTE 0 RESULTS.
09:07:17---READING AUDIO DATA... DONE! READ 5 CHUNKS.
09:07:17---Date-time: 2022-07-15 08:10:12
09:08:02---0.0;3.0-('Setophaga chrysoparia_Golden-cheeked Warbler', 0.02291845)
09:08:02---3.0;6.0-('Setophaga chrysoparia_Golden-cheeked Warbler', 0.024959052)
09:08:02---6.0;9.0-('Coragyps atratus_Black Vulture', 0.020198634)
09:08:02---9.0;12.0-('Vireo atricapilla_Black-capped Vireo', 0.030400226)
09:08:02---12.0;15.0-('Setophaga chrysoparia_Golden-cheeked Warbler', 0.021990024)
09:08:03---RECORDING_LENGTH set to 15
09:08:03---BirdNET-Pi/birdnet/bin/python3 BirdNET-Pi/scripts\analyze.py --i BirdSongs/July-2022/15-Friday/2022-07-15-birdnet-08:10:27.wav --o BirdSongs/July-2022/15-Friday/2022-07-15-birdnet-08:10:27.wav.csv --lat 30.XX --lon -95.XX --week 27 --overlap 0.0 --sensitivity 1.25 --min_conf 0.7 --exclude_list BirdNET-Pi/exclude_species_list.txt --birdweather_id "IN_USE"
09:08:06---ANALYZING AUDIO... DONE! Time 45.2 SECONDS
09:08:06---WRITING RESULTS TO BirdSongs/July-2022/15-Friday/2022-07-15-birdnet-08:10:12.wav.csv ... DONE! WROTE 0 RESULTS.
09:08:28---READING AUDIO DATA... DONE! READ 5 CHUNKS.
09:08:28---Date-time: 2022-07-15 08:10:27
09:09:36---0.0;3.0-('Ictinia mississippiensis_Mississippi Kite', 0.46421078)
09:09:36---3.0;6.0-('Vireo atricapilla_Black-capped Vireo', 0.025796736)
09:09:36---6.0;9.0-("Vireo bellii_Bell's Vireo", 0.028422138)
09:09:36---9.0;12.0-('Passerina ciris_Painted Bunting', 0.021404026)
09:09:36---12.0;15.0-('Setophaga chrysoparia_Golden-cheeked Warbler', 0.025952462)
09:09:38---RECORDING_LENGTH set to 15
09:09:38---BirdNET-Pi/birdnet/bin/python3 BirdNET-Pi/scripts\analyze.py --i BirdSongs/July-2022/15-Friday/2022-07-15-birdnet-08:10:43.wav --o BirdSongs/July-2022/15-Friday/2022-07-15-birdnet-08:10:43.wav.csv --lat 30.XX --lon -95.XX --week 27 --overlap 0.0 --sensitivity 1.25 --min_conf 0.7 --exclude_list BirdNET-Pi/exclude_species_list.txt --birdweather_id "IN_USE"
09:09:44---ANALYZING AUDIO... DONE! Time 66.1 SECONDS
09:09:44---WRITING RESULTS TO BirdSongs/July-2022/15-Friday/2022-07-15-birdnet-08:10:27.wav.csv ... DONE! WROTE 0 RESULTS.


htop results here - nothing unusual, but load average is high for sure:
![image](https://user-images.githubusercontent.com/6558981/179241735-5e3fc237-7839-4fa5-a393-2f48f886fcb8.png)

**Reply by mcguirepr89** [reply]
Hi, @grajohnt -- Out of curiosity, is anything else installed other than BirdNET-Pi?

**Reply by ehpersonal38** [reply]
I feel like that load average is way too high and is probably responsible for the spikes you're seeing. For reference on my Pi 4B I have a load average of about 1.

Also is it normal to have multiple server.py's running? They seem to be all using a lot of CPU as well. @mcguirepr89

**Reply by grajohnt** [reply]
No - this is vanilla BirdNET-Pi with the latest updates.

**Reply by grajohnt** [reply]
I should say - The only modification is that for the USB microphone I'm using, I had to add `dwc_otg.fiq_fsm_mask=0x5` to `/boot/cmdline.txt` to keep it from throwing errors. But I don't _think_ that would cause any issues here.

It may be that a 3B+ just isn't quite sufficient to keep up. I wonder if perhaps temperature throttling is causing the issue. After a reboot, load average drops to a more reasonable ~1.9.

**Reply by mcguirepr89** [reply]
>Also is it normal to have multiple server.py's running? They seem to be all using a lot of CPU as well.

That is unusual to me -- I'm not seeing that on my installation -- do you see that on yours, @ehpersonal38? Even if threading is enabled, that's still just one process??? So that is alarming -- I suppose I will need to run a clean installation and see if I can replicate this :(

**Reply by ehpersonal38** [reply]
There are about 5-6 server.py instances at the very bottom of htop, with low mem usage. They don't show up in "ps aux" though so I guess they're old/dead?
![putty_uNooTjILZq](https://user-images.githubusercontent.com/103586016/179265830-d7bd1ed1-2acd-471b-8d30-d8f74320beeb.png)

**Reply by mcguirepr89** [reply]
`htop` must show threads then -- `top` doesn't seem to show that (without the `-H` flag)

How many do you see when you issue this:

```
servers="$(pgrep python3|tr '\n' ',')"
top -p ${servers::-1} -c
```

**Reply by mcguirepr89** [reply]
@grajohnt You can see if your system has any messages from the kernel about throttling by going to "Tools" > "System Info". Near the bottom there is a "Viewer" widget where you'll see a section for the `vcgencmd stats`. It should have some indications like what you see below, but may indication errors:

```
..................................`vcgencmd stats`..............................
throttled=0x0
Binary: 0
```

**Reply by ehpersonal38** [reply]
> How many do you see when you issue this:

I see just 1 here!

**Reply by grajohnt** [reply]
This is what I see:

throttled=0x80008

Binary: 10000000000000001000
10000000000000001000
ISSUES DETECTED
Soft temperatue limit active
Soft temperature limit has occurred

**Reply by grajohnt** [reply]
...but it's running normally right now - I'll keep an eye on it

**Reply by grajohnt** [reply]
OK - scratch that theory...

It's running behind now, and not showing that it's throttled, and temp is at 58.5C, so no reason to do so.

Interesting, it's now stuck in a state where it's still recording audio, but not processing it.

Here's htop:
![image](https://user-images.githubusercontent.com/6558981/179323813-8399caf9-f35d-4f49-8c3c-0185614dc474.png)

It looks like that ffmpeg job is what's holding things up?

**Reply by grajohnt** [reply]
OK - Final update here - I upgraded to a Pi4 (same SD card), and have zero problems now and don't need reboots.

I'd perhaps suggest that you recommend that you only use a Pi4 for this. 

It's a great project, and much appreciated - thank you!

**Comment by nilspupils** [comment]
So it also happend to me, system was stuck with a "file not found" error looking for a wav. After deleting the last remaining wav file and rebooting twice, system is happily catching up on yesterday's files.

**Comment by GeekNJ** [comment]
I also ran into this for the first time after running my pi 4 for over 3 weeks. Here are my observations, but if there is any analysis that would be helpful, please let me know.

I noticed it was not displaying any new detections on the overview screen so I rebooted. I did not look at any system info before rebooting. 

When it came back online after the first reboot, I did a View log inside the app and saw it was now processing recordings from the previous evening. It took quite a while to catch up. It looked like about 3 seconds for each 15 second audio so that is 12 seconds for every minute it hadn't processed or 12 minutes to process each hour of backlog on my pi 4. 

After it finally caught up, I noticed that though the log seemed to indicate it was processing as normal, and there were still birds being recorded as I watched the live & spectogram pages, the analysis was showing 50% or less for 20 mins or more though I could hear unique birds just fine. So I rebooted again but needed to head out to dinner so I didn't stare at the pages. After the reboot post catch up, everything seemed to return to normal.

It could have been a coincidence but I usually have 5-25 detections in any given daylight hour and I could see then outside and hear in the recordings, but they just weren't hitting the confidence threshold. They were identified correctly but just a 50% or lower confidence.

One other observation is the Reboot button in Tools / System Controls doesn't seem to work for me. I click it, and it seems to prevent access to the app, but it never seems to fully reboot. If I run a ping to the IP address, it will just show a timeout. If I pull the power and restart, it always boots up in under 40 seconds. Next time I can also try to ssh in from my computer and do a sudo shutdown -r now to see how that works out.

**Reply by jflyons** [reply]
Hi GeekNJ - 

This was a helpful post for me even if unresolved, as I have been running for almost two months and it seems the more I understand about the system (and yes, tweek it) the worse it performs on detections. I do not think mine has been running behind like yours, but the numbers I see in the View Log are very low. It was suggested I could lower the thresholds but (for example) I just heard a House Finch sing from inside on the computer, it shows up on the spectrogram, but my log shows mostly "Dusky Grouse" around .10-.14, a few "Dusky Flycatcher" in the same range for good measure. so no House Finch at all. We've had huge numbers of Yellow-rumped Warblers and I haven't had a single one break the .7 threshold - most are .3 or below. 

Wonder if you ever got things back to even keel? I have repeatedly worked on the capture end with a variety of microphones and attempts at the correct settings. But I wonder if there is something I am missing, as I end up back on my original crappy little salvaged USB mic.

My last detection was aforementioned HOFI, last evening. Missed my nightly WESOs.

**Reply by ehpersonal38** [reply]
Try uploading a low confidence recording to the new model here and see what result you get - https://birdnet.cornell.edu/api/

**Reply by mcguirepr89** [reply]
Hi @jflyons 

There is an extremely important parameter that seems to be missing from your understanding of the analysis, which is the _`WEEK`_ parameter that references the eBird grid data from 2018 to determine the confidence score for the audio clip. As the weeks progress, the confidence score for a particular sound clip will change. That is, an audio clip of a Ruby Throated Humming bird might receive a 98% confidence score during week 30, and a 40% confidence score in week 2 using the same audio clip -- this is because the likelihood of a Ruby-Throated Humming bird is significantly less during week 2 compared to week 40 (that is just an example to illustrate how the `WEEK` parameter affects detections.

I can assure you that there is nothing in the system that degrades overtime and is somehow replenished from a reboot -- that is at best a coincidence.

I hope that extra piece of very important information increases your understanding of the system and helps you in determining what is happening during your observations.

My best,
Patrick

**Reply by jflyons** [reply]
Yes that was missing from my understanding, not logically but needed it spelled out I guess! The message regarding the YR Warblers needs to be updated as the detections (over .7) have really mounted up today, as the birds linger. Back to the RUHU example, did the detection algorithm with the week variable changed make the rating go from 45% to 75% with my local yellow-rumps?

This is all very interesting and thanks for explaining patiently. I love this tool and am committed to getting the most out of it for my local population studies. We tend to see what we want to see, and I think many of us expect our tech solutions to degrade so perhaps I will blame that. And of course the reboot is a bit like throwing that pinch of salt over your shoulder or other superstitions!

**Reply by jflyons** [reply]
> Try uploading a low confidence recording to the new model here and see what result you get - https://birdnet.cornell.edu/api/

Thanks @ehpersonal38 - one thing I haven't figured out is how to snag a recording from a non-detected sample, i.e. under .7 threshold?

**Reply by ehpersonal38** [reply]
Raw recordings are saved to `~/BirdSongs/Processed`. It only keeps the most recent 100 though so you'll have to be quick in snagging them!

**Reply by jflyons** [reply]
A related question? How (or is it possible) does one capture a recording on-demand, if it can? Much like the BirdNET app I am thinking, be able to copy a section of just-heard input, and then analyze? Thanks - just thought I'd put it out there!
